### PR TITLE
[quant][fx] Using native backend_config_dict in prepare

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75521
* #75520
* __->__ #75469

Summary:
Previously we are still relying on the registration mechnism and get the default quantize handlers that are registered,
now we have moved all registration to backend_config_dict we can get all quant patterns just from backend_config_dict now.

This PR enables using native backend_config_dict everywhere in prepare when the backend_config_dict is None, we'll also
do similar changes in convert as well

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35487228](https://our.internmc.facebook.com/intern/diff/D35487228)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35487228).

File: test/quantization/fx/test_numeric_suite_fx.py [hunk 1]
```diff
@@ -815,10 +815,12 @@ def _test_extract_weights(
         for extract_weights_fun in (extract_weights, _extract_weights_impl):
             # test both m vs mp and mp vs mq
             for m1, m2 in ((m, mp), (mp, mq)):
+                print("m1 m2:", m1, m2)
```
Comment (vkuzo) [inline test/quantization/fx/test_numeric_suite_fx.py:818]: nit: remove

File: test/quantization/fx/test_numeric_suite_fx.py [hunk 2]
```diff
@@ -815,10 +815,12 @@ def _test_extract_weights(
...
                 results = extract_weights_fun('a', m1, 'b', m2)
                 self.assertTrue(
                     len(results) == results_len,
                     f"expected len {results_len}, got len {len(results)}")
+                print("results:", results)
```
Comment (vkuzo) [inline test/quantization/fx/test_numeric_suite_fx.py:823]: nit: remove

File: torch/ao/quantization/fx/prepare.py [hunk 3]
```diff
@@ -145,11 +143,23 @@ def is_input_arg_dtype_supported_by_backend(
...
     if is_activation:
-        input_activation_dtype = dtype_config.get("input_activation_dtype", None)
-        return input_activation_dtype is None or \
-            node_name_to_target_dtype[node.name]["input_activation_dtype"] == input_activation_dtype
+        is_dynamic = dtype_config.get("is_dynamic", False)
+        if is_dynamic:
+            input_activation_dtype = dtype_config.get("input_dtype", None)
+            # TODO: change this after the is_dynamic refactor is landed
+            compute_dtype = node_name_to_target_dtype[node.name].get("input_activation_compute_dtype", None)
+            # print("compute dtype:", compute_dtype)
```
Comment (vkuzo) [inline torch/ao/quantization/fx/prepare.py:151]: nit: remove this and other commented out prints

**Review by vkuzo — APPROVED** [review]
lgtm!

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35487228).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35487228).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35487228).

**Pull request closed without merging**